pandas/core/series.py

```diff
@@ -2002,8 +2002,25 @@ def diff(self, periods=1):
         5    NaN
         dtype: float64
         """
-        result = algorithms.diff(com._values_from_object(self), periods)
-        return self._constructor(result, index=self.index).__finalize__(self)
+        def _diff(x):
+            result = algorithms.diff(com._values_from_object(self), x)
+            return self._constructor(result, index=self.index).__finalize__(self)
+
+        if isinstance(periods, int):
+            return _diff(periods)
+        elif isinstance(periods, list):
+            if len(periods) == 0:
+                raise ValueError('Must provide non empty list to periods')
+            else:
+                result = _diff(periods[0]).to_frame().add_suffix('{}{}'.format(suffix, periods[0]))
+                for period in periods[1:]:
+                    next_addition = _diff(period).to_frame().add_suffix('{}{}'.format(suffix, period))
+                    result = result.join(next_addition)
```

Comment:
This is doing `len(periods)` joins. Better to build up a list of frames and then concat.

Reply:
I didn't want to call concat b/c it raised the possibility of circular imports (I *think* the file in which pd.concat is defined also imports pd.Series)